collection.go

```diff
@@ -176,7 +176,7 @@ func (c *Collection) add(key string, exp Exp, val []byte, isJSON bool) (added bo
 
 func (c *Collection) SetRaw(key string, exp Exp, opts *sgbucket.UpsertOptions, val []byte) (err error) {
 	traceEnter("SetRaw", "%q, %d, ...", key, exp)
-	err = c.set(key, exp, opts, val, false)
+	err = c.set(key, exp, opts, val, json.Valid(val))
```

Comment:
The gocb setRaw will always end up sending the data type as gocbcore.BinaryType, based on RawBinaryEncoder.Encode().

Are you thinking that Couchbase Server ignores those SDK flags and sets the data type as JSON anyway, if it's valid JSON?